tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import utils
+
+
+def test_http_get_all(test_env, default_request, tmpdir):
+    """
+    Check the API endpoint to get all requests.
+
+    Process:
+    * Pre-populate data in Cachito by submitting basic requests and waiting for them to complete.
+    * Verify the submitted requests are not incorrectly marked as "in_progress".
+    * Verify requests can be found using by using the state=complete filter.
+    Checks:
+    * Check that response code is 200
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    for _ in range(requests_amount):
+        initial_response = client.create_new_request(
+            payload={
+                "repo": test_env["package"]["repo"],
+                "ref": test_env["package"]["ref"],
+                "pkg_managers": test_env["package"]["pkg_managers"],
+            },
+        )
+        completed_response = client.wait_for_complete_request(initial_response)
```

Comment:
Optional: This would make things a little faster if the Cachito instance has multiple workers.
```suggestion
    payload = {
        "repo": test_env["package"]["repo"],
        "ref": test_env["package"]["ref"],
        "pkg_managers": test_env["package"]["pkg_managers"],
    }
    for _ in range(requests_amount):
        response_one = client.create_new_request(payload)
        response_two = client.create_new_request(payload)
        completed_response = client.wait_for_complete_request(response_one)
        completed_response_two = client.wait_for_complete_request(response_two)
```

Reply:
In that case, I don't feel we need for loop. I changed it like: 

```
    payload = {
        "repo": test_env["package"]["repo"],
        "ref": test_env["package"]["ref"],
        "pkg_managers": test_env["package"]["pkg_managers"],
    }
    for _ in range(requests_amount):
        initial_response = client.create_new_request(payload)
        completed_response = client.wait_for_complete_request(initial_response)
        response_data = completed_response.data
        submitted_requests.append(response_data["id"])
```